File: fyn-flutter-app/lib/features/message/data/models/conversation_type.dart
enum ConversationType {
  direct,
  group,
  groupMeetup,
  friendsGroup;

  static ConversationType fromString(String? value) {
    switch (value?.toUpperCase()) {
      case 'GROUP_MEETUP':
        return ConversationType.groupMeetup;
      case 'FRIENDS_GROUP':
        return ConversationType.friendsGroup;
      case 'GROUP':
        return ConversationType.group;
      case 'DIRECT':
      default:
        return ConversationType.direct;
    }
  }

  String get serverValue {
    switch (this) {
      case ConversationType.groupMeetup:
        return 'GROUP_MEETUP';
      case ConversationType.friendsGroup:
        return 'FRIENDS_GROUP';
      case ConversationType.group:
        return 'GROUP';
      case ConversationType.direct:
      default:
        return 'DIRECT';
    }
  }

  bool get isGroup => this != ConversationType.direct;
}

File: fyn-monolithic/src/main/java/com/fyn_monolithic/model/message/ConversationType.java
package com.fyn_monolithic.model.message;

public enum ConversationType {
    DIRECT, // 1-on-1 chat between two users
    GROUP, // Legacy - kept for backward compatibility
    GROUP_MEETUP, // Group chat for meetup (auto-created when meetup has accepted members)
    FRIENDS_GROUP // User-created friend group chat
}
